docs/scalars_and_keras.ipynb

```diff
@@ -440,28 +425,374 @@
+        "  \n",
+        "  def train_step(self, data):\n",
+        "    x, y = data\n",
+        "    with tf.GradientTape() as tape:\n",
+        "      y_pred = self.model(x, training=True)\n",
+        "      loss = self.compiled_loss(y, y_pred)\n",
+        "      mse = tf.keras.losses.mean_squared_error(y, K.max(y_pred, axis=-1))\n",
+        "    self.optimizer.minimize(loss, self.trainable_variables, tape=tape)\n",
+        "    with train_writer.as_default(step=self._train_counter):\n",
+        "      tf.summary.scalar('batch_loss', loss)\n",
+        "      tf.summary.scalar('batch_mse', mse)\n",
+        "    return self.compute_metrics(x, y, y_pred, None)\n",
+        "  \n",
+        "  def call(self, x):\n",
+        "    x = self.model(x)\n",
+        "    return x\n",
+        "\n",
+        "# Adds custom batch-level metrics to our previous Functional API model\n",
+        "model = MyModel(create_model())\n",
+        "model.compile(optimizer='adam',\n",
+        "              loss='sparse_categorical_crossentropy',\n",
+        "              metrics=['accuracy'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IlqPZ-h_1_0z"
+      },
+      "source": [
+        "Setting up a summary writer to a different log directory:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Zjv_7dRwiyf9"
+      },
+      "outputs": [],
+      "source": [
+        "log_dir = 'logs/batch_level/' + datetime.now().strftime(\"%Y%m%d-%H%M%S\") + '/train'\n",
+        "train_writer = tf.summary.create_file_writer(log_dir)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8wqqNaXP2bLc"
+      },
+      "source": [
+        "Define our Tensorboard callback to log both epoch-level and batch-level metrics to our log directory and call `model.fit()` with our selected `batch_size`: "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "YXK-iE0e2UOE"
+      },
+      "outputs": [],
+      "source": [
+        "tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir)\n",
+        "\n",
+        "model.fit(x=x_train, \n",
+        "          y=y_train,\n",
+        "          epochs=5,\n",
+        "          batch_size=500, \n",
+        "          validation_data=(x_test, y_test), \n",
+        "          callbacks=[tensorboard_callback])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "SsxwF83h2kiM"
+      },
+      "source": [
+        "Open Tensorboard with the new log directory and see both the epoch-level and batch-level metrics:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "XlcafPNY2oUW"
+      },
+      "outputs": [],
+      "source": [
+        "%tensorboard --logdir logs/batch_level"
```

Comment:
Optional, but ideally, we would include a screenshot that shows what this command output would look like - this would be rendered in the HTML version of the notebook shown on TensorFlow.org.  There's examples above like this:

```
<!-- <img class=\"tfo-display-only-on-site\" src=\"https://github.com/tensorflow/tensorboard/blob/master/docs/images/scalars_custom_lr.png?raw=1\"/> -->
```

We'd need to add another image to https://github.com/tensorflow/tensorboard/tree/master/docs/images and then reference in here in a similar way.  This could also be done later, though.

Reply:
This can be done later, as publishing this PR sooner would help with a lot of recent YAQS questions on this issue. Thanks!